### PR TITLE
Implement Drain Life skill for summoner class

### PR DESCRIPTION
**Pull request by bernatvadell**
I have implemented the Drain Life skill for the Summoner class.

To do this I had to modify several classes in order to propagate the damage done to the skillarea plugin.

Please take a look at it.

### CODE REVIEW

File: src/GameLogic/IAttackable.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public interface IAttackable : IIdentifiable, ILocateable
     /// <param name="skill">The skill.</param>
     /// <param name="isCombo">If set to <c>true</c>, the attacker did a combination of skills.</param>
     /// <param name="damageFactor">The damage factor.</param>
-    ValueTask AttackByAsync(IAttacker attacker, SkillEntry? skill, bool isCombo, double damageFactor = 1.0);
+    ValueTask<HitInfo?> AttackByAsync(IAttacker attacker, SkillEntry? skill, bool isCombo, double damageFactor = 1.0);
```
Comment (sven-n) [inline src/GameLogic/IAttackable.cs:68]: Please add returns-comment

**Review by sven-n — CHANGES_REQUESTED** [review]

File: src/GameLogic/IAttackable.cs (hunk 1 above)
Comment (bernatvadell) [reply src/GameLogic/IAttackable.cs:69]: done!

**Comment by bernatvadell** [comment]
Realted task: [214 - Drain Life](https://github.com/orgs/MUnique/projects/3#)

File: src/Persistence/Initialization/VersionSeasonSix/SkillsInitializer.cs [hunk 2]
```diff
@@ -171,7 +171,7 @@ public override void Initialize()
         this.CreateSkill(SkillNumber.SpellofRestriction, "Spell of Restriction", CharacterClasses.All, distance: 3, manaConsumption: 30, elementalModifier: ElementalType.Ice, cooldownMinutes: 5);
         this.CreateSkill(SkillNumber.SpellofPursuit, "Spell of Pursuit", CharacterClasses.All, manaConsumption: 30, elementalModifier: ElementalType.Ice, cooldownMinutes: 10);
         this.CreateSkill(SkillNumber.ShieldBurn, "Shield-Burn", CharacterClasses.All, distance: 3, manaConsumption: 30, elementalModifier: ElementalType.Ice, cooldownMinutes: 5);
-        this.CreateSkill(SkillNumber.DrainLife, "Drain Life", CharacterClasses.AllSummoners, DamageType.Curse, 35, 6, manaConsumption: 50, energyRequirement: 150);
+        this.CreateSkill(SkillNumber.DrainLife, "Drain Life", CharacterClasses.AllSummoners, DamageType.Curse, 35, 6, manaConsumption: 50, energyRequirement: 150, skillType: SkillType.AreaSkillExplicitTarget);
```
Comment (sven-n) [inline src/Persistence/Initialization/VersionSeasonSix/SkillsInitializer.cs:174]: We'll need an IConfigurationUpdatePlugIn for that to change existing configurations

Comment (bernatvadell) [reply src/Persistence/Initialization/VersionSeasonSix/SkillsInitializer.cs:174]: done

**Review by sven-n — APPROVED** [review]

**Comment by bernatvadell** [comment]
@sven-n can you close this task? https://github.com/orgs/MUnique/projects/3/views/1?pane=issue&itemId=67145122

**Pull request merged**